### PR TITLE
Update match id on post save

### PR DESCRIPTION
**Pull request by adamledwards**
## Description of change

This PR add the functionality to update the match id on post save. 
The celery task is triggered by a post save signal attached to the win models. 
This give it the ability to be triggered if its updated from any channel e.g front end or django admin.

## Backend stack changes required.
This PR requires redis to be changed from cluster mode to stand alone as celery does not support cluster mode.

### CODE REVIEW

File: wins/tasks/match_id_task.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+@shared_task(
+    bind=True,
+)
+def update_match_id(self, win_id):
+    """Get match id from compmay matching service and save in to the model."""
+    win = Win.objects.get(pk=win_id)
+    try:
+        response = get_match_ids([win])
+        match_id = _extract_match_id(response.json())
```
Comment (sekharpanja) [inline wins/tasks/match_id_task.py:43]: We don't need to update the win if `match_id` is None, right? Why don't we check if its `None` and skip saving the win.

File: wins/tasks/match_id_task.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        win = Win.objects.get(pk=win_id)
```
Comment (sekharpanja) [inline wins/tasks/match_id_task.py:44]: This has already been done one line 40. So keep one of them.

File: wins/tests/test_task.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+from rest_framework.status import HTTP_200_OK
+
+from wins.company_matching_utils import CompanyMatchingServiceException
+from wins.factories import WinFactory
+from wins.tasks.match_id_task import update_match_id
+from test_helpers.hawk_utils import HawkMockJSONResponse
+
+
+@pytest.mark.django_db
+class TestMatchIdTask:
```
Comment (sekharpanja) [inline wins/tests/test_task.py:19]: You might want to add a test case when match id is None.

File: wins/signals.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+from wins.tasks.match_id_task import update_match_id
+from wins.models import Win
+
+logger = logging.getLogger(__name__)
+
+
+@receiver(post_save, sender=Win, dispatch_uid='win_post_save_dispatch')
+def save_match_id(sender, instance, **kwargs):
+    """Save match id on a post save signal."""
+    update_fields = kwargs['update_fields']
```
Comment (sekharpanja) [inline wins/signals.py:16]: You could do this perhaps:
`update_fields = kwargs.get('update_fields', [])`
And avoid if condition.

File: wins/tasks/match_id_task.py (hunk 1 above)
Comment (adamledwards) [reply wins/tasks/match_id_task.py:43]: Because match_id can change I assume this go for a company have a match_id then later it doesn't
e.g
User enters a wrong detail that generates a match that appears in the wrong company.
User corrects their change and there is actually not a match.
This logic allows incorrect matches to be taking out of a company.

File: wins/tasks/match_id_task.py (hunk 2 above)
Comment (adamledwards) [reply wins/tasks/match_id_task.py:44]: cheers

File: wins/signals.py (hunk 4 above)
Comment (adamledwards) [reply wins/signals.py]: This actually won't work because 
`update_fields = kwargs.get('update_fields', [])` will return the default if the field is missing.
However  save_match_id is explicitly being passed None.
